### ISSUE TITLE
Training error and testing error increase with the epoch 

### ISSUE THREAD

**Issue by waldstein1983** [ibm-research-tokyo/dybm#7]
Dear authors,
I tested this code by my own uni-variant time series data. 
--------Log---------------
Number of Observations/datapoints in each dimension 418
Train data X shape:  (362, 1, 1)
Train data Y shape:  (362, 1)
Test data X shape:  (44, 1, 1)
Test data Y shape:  (44, 1)
Vali data X shape:  (6, 1, 1)
Vali data Y shape:  (6, 1)
Backend Qt5Agg is interactive backend. Turning interactive mode on.
*****************************************
Learning with DyBM model RNNGaussian
*****************************************
learning with delay = 3
train error: 0.36476
test error: 0.10549 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.105494
train error: 0.13199
test error: 0.08342 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.094458
train error: 0.12146
test error: 0.08052 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.089813
train error: 0.11963
test error: 0.07742 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.086715
train error: 0.12486
test error: 0.08193 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.085758
train error: 0.13522
test error: 0.10800 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.089465
train error: 0.15324
test error: 0.16453 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.100189
train error: 0.17873
test error: 0.24287 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.118025
train error: 0.20790
test error: 0.32390 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.140900
train error: 0.23831
test error: 0.40343 
delay = 3.000000 decay = [0.5] curr_mean_error = 0.167152
----------Log----------
We can see that the errors increase much.
Can you please help to identify the problem?
Thx

**Comment by osogami** [comment]
It is most likely that your learning rate is too large.  Could you try something like set_learning_rate(0.001)?

**Comment by waldstein1983** [comment]
Thanks for your reply.
I used CompareLSTMandDyBM.py in benchmark folder and learning rate is already set to 0.001.
Please check line 275-292

**Comment by osogami** [comment]
I see.  So, the training error increases with RNNGaussian but decreases with LinearDyBM or Gaussian?  If you want a model similar to RNNGaussian but has monotonic decrease in the training error, you can try LinearDyBM with ESN (https://github.com/ibm-research-tokyo/dybm/blob/master/src/pydybm/time_series/dybm.py#L72).
